src/Files.App/Commands/Manager/CommandManager.cs

```diff
@@ -563,5 +577,112 @@ private void Action_PropertyChanged(object? sender, PropertyChangedEventArgs e)
 				}
 			}
 		}
+
+		[DebuggerDisplay("Command {Code} (Modifiable)")]
+		private class ModifiableCommand : ObservableObject, IRichCommand
+		{
+			public event EventHandler? CanExecuteChanged;
+
+			private IRichCommand BaseCommand;
+			private ImmutableDictionary<KeyModifiers, IRichCommand> ModifiedCommands;
+
+			public CommandCodes Code => BaseCommand.Code;
+
+			public string Label => BaseCommand.Label;
+			public string LabelWithHotKey => BaseCommand.LabelWithHotKey;
+			public string AutomationName => BaseCommand.AutomationName;
+
+			public string Description => BaseCommand.Description;
+
+			public RichGlyph Glyph => BaseCommand.Glyph;
+			public object? Icon => BaseCommand.Icon;
+			public FontIcon? FontIcon => BaseCommand.FontIcon;
+			public Style? OpacityStyle => BaseCommand.OpacityStyle;
+
+			public bool IsCustomHotKeys => BaseCommand.IsCustomHotKeys;
+			public string? HotKeyText => BaseCommand.HotKeyText;
+
+			public HotKeyCollection HotKeys
+			{
+				get => BaseCommand.HotKeys;
+				set => BaseCommand.HotKeys = value;
+			}
+
+			public bool IsToggle => BaseCommand.IsToggle;
+
+			public bool IsOn
+			{
+				get => BaseCommand.IsOn;
+				set => BaseCommand.IsOn = value;
+			}
+
+			public bool IsExecutable => BaseCommand.IsExecutable;
+
+			public ModifiableCommand(IRichCommand baseCommand, Dictionary<KeyModifiers, IRichCommand> modifiedCommands)
+			{
+				BaseCommand = baseCommand;
+				ModifiedCommands = modifiedCommands.ToImmutableDictionary();
+
+				if (baseCommand is ActionCommand actionCommand)
```

Comment:
Invert if to reduce nesting.

Reply:
Since this code also serves as a cast, I left it as is.